department-of-veterans-affairs/vets-api
# [64196] - Create TermsOfUseAgreement model

## Summary
- Create `TermsOfUseAgreement` model. Migrations will be split out into a [separate PR](https://github.com/department-of-veterans-affairs/vets-api/pull/13632)

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#64196
- department-of-veterans-affairs/vets-api#13632

## How to test
- Checkout and run migrations from here - department-of-veterans-affairs/vets-api#13632
  - Migrate database - `rails db:migrate`
- In rails console create an agreement from a user_account. - `rails c`
  ```ruby
   UserAccount.last.terms_of_use_agreements.create(agreement_version: 'v1', response: :declined)
   UserAccount.last.terms_of_use_agreements.create(agreement_version: 'v1', response: :accepted)
  ```
- The user should have 2 TermsOfUseAgreements, one declined and another accepted 

  ```ruby
   UserAccount.last.terms_of_use_agreements
  ```

## What areas of the site does it impact?
Terms of use

## Acceptance criteria
